app/views/service_worker/index.js.erb

```diff
@@ -1,91 +1,21 @@
 // Service worker file. Renders as serviceworker.js
+// This is now a self-destructing service worker. See https://github.com/NekR/self-destroying-sw
 
-<% unless Rails.env.test? || ENV["SKIP_SERVICEWORKERS"] == "true" %>
-  const VERSION = "1.0" // Increment to invalidate old assets.
-  const OFFLINE_URL = "offline.html";
-  const OFFLINE_NAME = OFFLINE_URL + VERSION
-
-  self.addEventListener("install", (event) => {
-    event.waitUntil(
-      (async () => {
-
-        // Setting {cache: 'reload'} in the new request will ensure that the
-        // response isn't fulfilled from the HTTP cache; i.e., it will be from
-        // the network.
-        const offlineCache = await caches.open(OFFLINE_NAME);
-        await offlineCache.add(new Request(OFFLINE_URL, { cache: "reload" }));
-      })()
-    );
-    // Force the waiting service worker to become the active service worker.
+<% unless Rails.env.test? %>
+  self.addEventListener('install', function(e) {
```

Comment:
The most reliable way to unregister a service worker is to create a new one that unregisters itself according to https://github.com/NekR/self-destroying-sw